camera/include/camera/Camera.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef RCBE_ENGINE_CAMERA_HPP
+#define RCBE_ENGINE_CAMERA_HPP
+
+#include <mutex>
+#include <memory>
+
+#include <data_types/math/euler_angles.hpp>
+#include <data_types/rendering/RenderingContext.hpp>
+#include <data_types/math/Transform.hpp>
+#include <data_types/math/Matrix.hpp>
+#include <data_types/math/Vector.hpp>
+
+namespace rcbe::rendering {
+class Camera {
+public:
+    using transform_type = math::Transform;
+
+    Camera() = delete;
+    Camera(const rendering::RenderingContextPtr &context, const math::Vector3d &position, const math::Vector3d &lookat, const math::Vector3d &up);
+    ~Camera() = default;
+
+    math::Matrix4x4 look_at(const math::Vector3d& camera_position, const math::Vector3d& lookat, const math::Vector3d& up);
+
+    [[nodiscard]]const transform_type& get_transform() const;
+
+    void reset_view();
+
+    [[nodiscard]] math::Vector3d camera_direction() const;
+    [[nodiscard]] const math::Vector3d &camera_up() const;
+    [[nodiscard]] math::Vector3d camera_right() const;
+
+    void translate(const rcbe::math::Vector3d& direction, const rcbe::math::Vector3d& look_direction);
```

Comment:
How about to give the scond arg a default value being equal to the first arg?

Reply:
The order of which args are evaluated is not defined, so no ))